Cargo.toml

```diff
@@ -11,6 +11,7 @@ members = [
     "internal-dns",
     "internal-dns-client",
     "nexus",
+    "nexus/src/authn",
     "nexus/src/authz/authz-macros",
     "nexus/src/db/db-macros",
```

Comment:
I know this existed before this PR - but for nested crates, what's the reason for putting things like `authn` under `nexus/src/authn` vs `nexus/authn`? I think we're inconsistent between, for example, Nexus and Oximeter.

(Personally, I by seeing a directory under `src`, my first impression is that it's a submodule, not a different crate. but maybe that doesn't matter)

Reply:
Done in e309665da2b81af3ec16b71721740f21e852e8a7